python/cudf/cudf/core/index.py

```diff
@@ -3032,3 +3037,43 @@ def __new__(
             )
 
         return as_index(data, copy=copy, dtype=dtype, name=name, **kwargs)
+
+
+def _concat_range_index(indexes: List[RangeIndex]) -> BaseIndex:
+    """
+    An internal Utility function to concat RangeIndex objects.
+    """
+    start = step = next_ = None
+
+    # Filter the empty indexes
+    non_empty_indexes = [obj for obj in indexes if len(obj)]
+
```

Comment:
I think the code would be clearer if you just handled the empty case up front:
```suggestion
    non_empty_indexes = [obj for obj in indexes if len(obj)]
    if not non_empty_indexes:
        return RangeIndex(0, 0)
```